lib/util.js

```diff
@@ -658,4 +659,38 @@ exports.getCompareAndLogEnvAndAgentProxies = function (agentOptions) {
 */
 exports.removeScheme = function (input) {
   return input.toString().replace(/(^\w+:|^)\/\//, '');
+};
+
+/**
+ * Checks if the provided file or directory permissions are correct.
+ * @param filePath
+ * @param expectedMode
+ * @param fsPromisesModule
+ * @returns {Promise<boolean>} resolves always to true for Windows
+ */
+exports.isFileModeCorrect = async function (filePath, expectedMode, fsPromisesModule) {
+  if (os.platform() === 'win32') {
+    return true;
+  }
+  const fsPromises = typeof fsPromisesModule !== 'undefined' ? fsPromisesModule : require('fs/promises');
```

Comment:
don't we always pass `fsPromiseModule`?

Reply:
We do pass it, however I thought that this way it will be easier to use in some cases, when developer will not provide a module to the function.

Do you think it would be better to use the `fsPromisesModule` straight away?

Reply:
I would make it mandatory parameter and woundn't care here about initialization

Reply:
Done

Reply:
Let's only name the variable `fsPromises` without the `Module` suffix

Reply:
Done